src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs

```diff
@@ -170,137 +118,81 @@ internal void DependencyRegen(WorkspaceModel ws)
         public WorkspaceDependencyView(WorkspaceDependencyViewExtension viewExtension,ViewLoadedParams p)
         {
             InitializeComponent();
-            DataContext = table;
             currentWorkspace = p.CurrentWorkspaceModel as WorkspaceModel;
             p.CurrentWorkspaceChanged += OnWorkspaceChanged;
             p.CurrentWorkspaceCleared += OnWorkspaceCleared;
             currentWorkspace.PropertyChanged += OnWorkspacePropertyChanged;
             loadedParams = p;
             dependencyViewExtension = viewExtension;
+            DependencyRegen(currentWorkspace);
+            packageManagerClientViewModel = loadedParams.PackageManagerClientViewModel;
         }
-    }
-
-    /// <summary>
-    /// Every table line
-    /// </summary>
-    public class Column
-    {
-        public ObservableCollection<ColumnData> ColumnsData
+        
+        /// <summary>
+        /// Send a request to the package manager client to download this package and its dependencies
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void DownloadPackage(object sender, RoutedEventArgs e)
         {
-            get; set;
+            var info = ((NodeLibraryDependencyRow)((Button)sender).DataContext).DependencyInfo;
+            var package = new PackageInfo(info.Name, info.Version);
+
+            packageManagerClientViewModel.InitiatePackageDownloadAndInstall(package);
+            DependencyRegen(currentWorkspace);
         }
     }
 
-    /// <summary>
-    /// Class defining data for each column
-    /// </summary>
-    public class ColumnData
+    public class NodeLibraryDependencyRow
```

Comment:
did something weird happen to this file? The diff is pretty hard to read, it looks like a bunch of public classes were deleted? Or am I mis-reading the diff?

Reply:
(you can't delete the public classes if they get released with another version - however short lived that version is) it's safer to just obsolete them as we don't know what binary compatibility issues this will cause without extensive testing, and it's just not worth it IMO.

Reply:
@mjkkirschner Do you remember the last case this was an issue? This class is only used for data binding to the UI so we can't make it internal. If any class went out with a beta release that we can no longer rename, that sounds like too strict..

Reply:
I'm not talking about beta releases.

If it's in 2.3 then we should consider it stable, unless @scottmitchell wants to refactor 2.3 before release to not include this class and then redo testing there.

Reply:
Sure, I guess I am wondering if the API restriction need to be applied on extension level. I understand once it is in RC branch we should make it stable but this dll is not in our public nuget so I do not expect developers to consume it nor us maintaining the stability.. My 2 cents..

Reply:
@QilongTang - that's a great great point! I think it's acceptable to make this entire .dll **not** part of the API.

👍 

I think we'll need to consider what consequences this has technically and also for build processes though.

Reply:
@mjkkirschner I don't think we have it specified according quickly glancing at https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions or https://github.com/DynamoDS/Dynamo/wiki/API-Changes. And seems we also missed 2.x versions' API changes...

Reply:
regarding the API Changes doc... I find it kind of useless work - the API diff job now computes this for us - or something like
https://www.fuget.org/packages/DynamoVisualProgramming.Core/2.2.1.5204/lib/net47/diff/2.1.0.7465/

is SOO much better than that page - it's only missing the explanations/readme portion.